core/pbcc_set_state.c

```diff
@@ -87,29 +89,32 @@ void pbcc_adjust_state(struct pbcc_context* core,
             bool end = false;
             do{
                 cg_temp = strchr(str_cg,',');
-                if (ch_cnt > 0 || cg_cnt > 0) { 
+                if ((ch_cnt > 0 || cg_cnt > 0) && json_state[mem_len - 1] != ',') { 
                     memcpy(json_state + mem_len, ",", cm_len);
                     mem_len += cm_len;
                 }
                 if (cg_temp == NULL) { end = true; cg_len = strlen(str_cg); }
                 else { cg_len = cg_temp - str_cg; }
 
-                if (cg_len == 0) { continue; }
+                if (cg_len != 0) { 
+                    char* curr_cg = (char*)malloc(cg_len + 1);
+                    memcpy(curr_cg, str_cg, cg_len);
+                    curr_cg[cg_len] = '\0';
 
-                char* curr_cg = (char*)malloc(cg_len + 1);
-                strcpy(curr_cg, str_cg);
+                    mem_len = json_kvp_builder(json_state, mem_len, curr_cg, (char*)state);
 
-                mem_len = json_kvp_builder(json_state, mem_len, curr_cg, (char*)state);
+                    free(curr_cg);
+                }
 
                 cg_cnt++;
                 str_cg = cg_temp + 1;
-                free(curr_cg);
             } while (false == end);
         }
 
         int cb_len = strlen("}");
-        memcpy(json_state + mem_len, "}", cb_len);
-        mem_len += cb_len;
+        int trim_comma = json_state[mem_len - 1] == ',' ? 1 : 0;
+        memcpy(json_state + mem_len - trim_comma, "}", cb_len);
+        mem_len += cb_len - trim_comma;
```

Comment:
If we want to do this, should it be like this:
```suggestion
        int trim_comma = json_state[mem_len - 1] == ',' ? strlen(",") : 0;
        memcpy(json_state + mem_len - trim_comma, "}", cb_len);
        mem_len += cb_len - trim_comma;
```
?